voicevox_engine/tts_pipeline/text_analyzer.py

```diff
@@ -358,6 +359,75 @@ def _utterance_to_accent_phrases(utterance: UtteranceLabel) -> list[AccentPhrase
     ]
 
 
+def _convert_to_hankaku(surface: str) -> str:
+    return surface.translate(
+        str.maketrans(
+            "".join(chr(0xFF01 + i) for i in range(94)),
+            "".join(chr(0x21 + i) for i in range(94)),
+        )
+    )
+
+
+c2k: e2k.C2K | None = None
+
+# e2kで変換するかどうか判断するルール
+# - 3文字以上
+# - 先頭が大文字/小文字、それ以降が小文字
+rule_convertible = re.compile(r"[a-zA-Z][a-z]{2,}")
+
+
+def extract_fullcontext_with_e2k(text: str) -> list[str]:
+    """e2kを用いて読みが不明な英単語をカタカナに変換し、フルコンテキストラベルを生成する"""
+    global c2k
+    njd_features: list[dict[str, Any]] = pyopenjtalk.run_frontend(text)
+    for i, feature in enumerate(njd_features):
+        # Mecabの解析で未知語となった場合、読みは空となる
+        # NJDは、読みが空の場合、読みを補完して品詞をフィラーとして扱う
+        if feature["pos"] != "フィラー" or feature["chain_rule"] != "*":
+            continue
+
+        if c2k is None:
+            c2k = e2k.C2K()
+
+        # OpenJTalkはアルファベットを全角に変換するので、半角に戻す
+        hankaku_string = _convert_to_hankaku(feature["string"])
+
+        if rule_convertible.fullmatch(hankaku_string):
+            kana = c2k(hankaku_string.lower())
+
+            # TODO: user_dict/model.py内の処理と重複しているため、リファクタリングする
+            rule_others = (
+                "[イ][ェ]|[ヴ][ャュョ]|[ウクグトド][ゥ]|[テデ][ィェャュョ]|[クグ][ヮ]"
+            )
+            rule_line_i = "[キシチニヒミリギジヂビピ][ェャュョ]|[キニヒミリギビピ][ィ]"
+            rule_line_u = "[クツフヴグ][ァ]|[ウクスツフヴグズ][ィ]|[ウクツフヴグ][ェォ]"
+            rule_one_mora = "[ァ-ヴー]"
+
+            njd_features[i] = {
+                "string": kana,
+                "pos": "名詞",
+                "pos_group1": "固有名詞",
+                "pos_group2": "一般",
+                "pos_group3": "*",
+                "ctype": "*",
+                "cform": "*",
+                "orig": feature["string"],
+                "read": kana,
+                "pron": kana,
+                "acc": 1,
+                "mora_size": len(
+                    re.findall(
+                        f"(?:{rule_others}|{rule_line_i}|{rule_line_u}|{rule_one_mora})",
+                        kana,
+                    )
+                ),
+                "chain_rule": "*",
+                "chain_flag": -1,
+            }
+
+    return pyopenjtalk.make_label(njd_features)  # type: ignore
+
+
 def text_to_accent_phrases(
     text: str,
     text_to_features: Callable[[str], list[str]] = pyopenjtalk.extract_fullcontext,
```

Comment:
ここの初期値なくしたほうが安全かも？
というTODOだけでもとりあえず良いかも！
```suggestion
    text_to_features: Callable[[str], list[str]] = pyopenjtalk.extract_fullcontext, # TODO: 初期値をなくす？
```